packages/helix-shared-storage/src/storage.js

```diff
@@ -400,42 +400,74 @@ class Bucket {
    * Remove object(s)
    *
    * @param {string|string[]} path source key(s)
+   * @param {string} [sourceInfo] informational message of the source
+   * @param {boolean} [stopOnError]
    * @returns result obtained from S3
    */
-  async remove(path) {
+  async remove(path, sourceInfo = '', stopOnError = false) {
+    const { log, bucket } = this;
+
     if (Array.isArray(path)) {
-      const input = {
-        Bucket: this.bucket,
-        Delete: {
-          Objects: path.map((p) => ({ Key: sanitizeKey(p) })),
-        },
+      // slice into chunks of MAX_DELETE_OBJECTS at most
+      const chunks = Array.from({
+        length: Math.ceil(path.length / MAX_DELETE_OBJECTS),
+      }, (v, i) => path.slice(i * MAX_DELETE_OBJECTS, i * MAX_DELETE_OBJECTS + MAX_DELETE_OBJECTS));
+
+      let oks = 0;
+      let errors = 0;
+      const result = {
+        Deleted: [],
+        Errors: [],
       };
-      // delete on s3 and r2 (mirror) in parallel
-      try {
-        const result = await this.sendToS3andR2(DeleteObjectsCommand, input);
-        this.log.info(`${result.Deleted.length} objects deleted from bucket ${input.Bucket}.`);
-        return result;
-      } catch (e) {
-        const msg = `removing ${input.Delete.length} objects from bucket ${input.Bucket} failed: ${e.message}`;
-        this.log.error(msg);
-        const e2 = new Error(msg);
-        e2.status = e.$metadata.httpStatusCode;
-        throw e2;
-      }
+      await processQueue(chunks, async (chunk) => {
+        log.debug(`deleting ${chunk.length} from ${bucket}`);
+        const input = {
+          Bucket: bucket,
+          Delete: {
+            Objects: chunk.map((p) => ({ Key: sanitizeKey(p) })),
+          },
+        };
+
+        try {
+          // delete on s3 and r2 (mirror) in parallel
+          const res = await this.sendToS3andR2(DeleteObjectsCommand, input);
+          if (res.Deleted) {
+            result.Deleted.push(...res.Deleted);
+          }
+          if (res.Errors) {
+            result.Errors.push(...res.Errors);
+            errors += res.Errors.length;
+          }
+          oks += chunk.length;
```

Comment:
still add this to `oks`, even when `res.Errors` is non empty?

Reply:
maybe we should update the oks and errors depending on the result